### PR TITLE
Question Service APIs

### LINKED ISSUE #8: Question Service - to maintain a question repository indexed by difficulty level
- Setup MongoDB for questions 
- CRUD endpoints API spec 
- ...

### PR DESCRIPTION
**Pull request by chunweii**
Fixes #8

- [X] Set up MongoDB
- [X] Create a question
- [x] Get questions by some fields
- [x] Get question by object id
- [x] Update a question by object id
- [x] Delete a question by object id
- [x] Get a random question by difficulty and topic

### CODE REVIEW

File: services/question-service/src/generate-mongodb-schema.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+const uri = process.env.MONGO_ATLAS_URL || 'mongodb://localhost:27017';
+
+// Create a MongoClient with a MongoClientOptions object to set the Stable API version
+const mongoClient = new MongoClient(uri, {
+  serverApi: {
+    version: ServerApiVersion.v1,
+    strict: true,
+    deprecationErrors: true,
+  }
+});
```
Comment (ong6) [inline services/question-service/src/generate-mongodb-schema.ts:14]: looks like we're not using Prisma for this?

**Review by ong6 — APPROVED** [review]
LGTM

File: services/question-service/src/generate-mongodb-schema.ts (hunk 1 above)
Comment (chunweii) [reply services/question-service/src/generate-mongodb-schema.ts:14]: Nope. But we can change this to use prisma later on

Comment (ong6) [reply services/question-service/src/generate-mongodb-schema.ts:14]: ok feel free to merge after fixing conflics

**Pull request merged**